connection_test.go

```diff
@@ -1422,3 +1422,48 @@ func TestCustomDialer(t *testing.T) {
 		assert.Equal(t, 1, customDialerCalledCount, "custom dialer used for establishing connection")
 	})
 }
+
+func TestInboundConnContext(t *testing.T) {
+	opts := testutils.NewOpts().NoRelay().SetConnContext(func(ctx context.Context, conn net.Conn) context.Context {
+		return context.WithValue(ctx, "foo", "bar")
+	})
+
+	testutils.WithTestServer(t, opts, func(t testing.TB, ts *testutils.TestServer) {
+		alice := ts.Server()
+		testutils.RegisterFunc(alice, "echo", func(ctx context.Context, args *raw.Args) (*raw.Res, error) {
+			// Verify that the context passed into the handler inherits from the base context
+			// set by ConnContext
+			assert.Equal(t, "bar", ctx.Value("foo"), "Value unexpectedly different from base context")
+			return &raw.Res{Arg2: args.Arg2, Arg3: args.Arg3}, nil
+		})
+
+		copts := testutils.NewOpts()
+		bob := ts.NewClient(copts)
+		err := testutils.CallEcho(bob, ts.HostPort(), ts.ServiceName(), nil)
+		assert.NoError(t, err, "Echo failed")
```

Comment:
```suggestion
		testutils.AssertEcho(t, bob, ts.HostPort(), ts.ServiceName())
```

same everywhere you do call + check for no error.